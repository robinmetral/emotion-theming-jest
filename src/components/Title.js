import React from "react"
import { css } from "@emotion/core"

const Title = ({ title }) => (
  <h1
    css={theme =>
      css`
        color: ${theme.colors.primary};
      `
    }
  >
    {title}
  </h1>
)

export default Title
