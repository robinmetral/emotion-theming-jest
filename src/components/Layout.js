import React from "react"
import { ThemeProvider } from "emotion-theming"

import Title from "./Title"

export const theme = {
  colors: {
    primary: "rebeccapurple",
  },
}

const Layout = ({ children }) => (
  <ThemeProvider theme={theme}>
    <Title title="emotion-theming-jest" />
    {children}
  </ThemeProvider>
)

export default Layout
