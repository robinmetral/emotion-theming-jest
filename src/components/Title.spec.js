import React from "react"
import renderer from "react-test-renderer"
import { ThemeProvider } from "emotion-theming"

import Title from "./Title"
import { theme } from "./Layout"

describe("Header", () => {
  it("renders correctly", () => {
    const tree = renderer
      .create(
        <ThemeProvider theme={theme}>
          <Title title="emotion-theming-jest" />
        </ThemeProvider>
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})
