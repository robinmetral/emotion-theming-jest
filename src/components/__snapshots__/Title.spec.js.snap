// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders correctly 1`] = `
.emotion-0 {
  color: rebeccapurple;
}

<h1
  className="emotion-0"
>
  emotion-theming-jest
</h1>
`;
